pandas/core/ops.py

```diff
@@ -862,10 +862,16 @@ def masked_arith_op(x, y, op):
         # mask is only meaningful for x
         result = np.empty(x.size, dtype=x.dtype)
         mask = notna(xrav)
+
         if mask.any():
             with np.errstate(all='ignore'):
                 result[mask] = op(xrav[mask], y)
 
+        if op == pow:
+            result = np.where(~mask, x, result)
```

Comment:
can you add the same comments as you have above here (e.g. 1 ** np.nan...)

Reply:
I reworked this to update the mask for both `pow` and `rpow`, rather than adjusting the result.